### ISSUE TITLE
Add custom certificates option

### ISSUE THREAD

**Issue by piggyvenus** [Microsoft/openshift-container-platform#28]
It would be nice to be able to use custom certificates instead of just using the self signed cert.

**Comment by haroldwongms** [comment]
@piggyvenus I agree.  I have been looking at incorporating this into the template as well as the Azure Marketplace offer.  Give me a little time to work on this as my calendar is completely swamped right now.

**Comment by piggyvenus** [comment]
Thanks for following up on this one!

**Comment by haroldwongms** [comment]
Update: custom certificates are being incorporated now for master web console and routing domain.  Stay tuned as we finish working through the last few bugs before the PR.

**Comment by linuxius** [comment]
Do you have a pre-release yet I could try. I found that changing the certificates after installation is quite impossible.

**Comment by haroldwongms** [comment]
@linuxius my working branch is here: https://github.com/haroldwongms/openshift-container-platform/tree/certandproxy.  I have updated the readme but may still be missing some documentation.

**Comment by linuxius** [comment]
Thanks for the speedy response. I cloned your working tree and adapted the  azuredeploy.parameters.json file according to the readme. But I am getting the following message:

"Unable to download deployment content from 'https://raw.githubusercontent.com/Microsoft/openshift-container-platform/master/nested/masterprep.json'. 
The tracking Id is 'b7e81cad-da4b-4b2f-97ba-f34d16d1f2ed'. Please see https://aka.ms/arm-deploy for usage details.\"

"Deployment template validation failed: 'The template parameters 'customMasterCertFile, customRoutingCertFile, noProxyEntry, masterCertType, routingCertType, 
customRoutingKeyFile, httpProxyEntry, proxySettings, httpsProxyEntry, customMasterKeyFile, customMasterCaFile, customRoutingCaFile' in the parameters file are 
not valid; they are not present in the original template and can therefore not be provided at deployment time. The only supported parameters for this template 
are...... "

I assume the "_artifactsLocation" is wrong for your working tree. I tried the following, but without success
"_artifactsLocation": {
  "value": "https://raw.githubusercontent.com/haroldwongms/openshift-container-platform/tree/certandproxy"
},


Can you tell me what I am doing wrong?

Thanks.

**Comment by haroldwongms** [comment]
@linuxius   Please update the _artifactsLocation parameter in the parameters file versus the azuredeploy.json file.  You do want to use https://raw.githubusercontent.com/haroldwongms/openshift-container-platform/tree/certandproxy as you stated above.

**Comment by linuxius** [comment]
I tried that but I am getting the following error in the deployment log:
"code": "InvalidContentLink", "message": "Unable to download deployment content from 'https://raw.githubusercontent.com/haroldwongms/openshift-container-platform/tree/certandproxy/nested/galleryclusternode.json'. The tracking Id is '2ec87ad2-080a-4250-9a16-e909f1778169'. Please see https://aka.ms/arm-deploy for usage details."

I assume the link is incorrect.

**Comment by haroldwongms** [comment]
My mistake.  Please use: https://raw.githubusercontent.com/haroldwongms/openshift-container-platform/certandproxy  (don't have tree at the end)
